libraries/reconstruction_cuda/cuda_gpu_iirconvolve.cu

```diff
@@ -1,107 +1,314 @@
-// inspired by https://devblogs.nvidia.com/efficient-matrix-transpose-cuda-cc/
-// TILE_DIM=32, BLOCK_ROWS=8
-// No bank-conflict transpose
-// Same as transposeCoalesced except the first tile dimension is padded
-// to avoid shared memory bank conflicts.
-// can be used to transpose non-square 2D arrays
-__global__
-void transposeNoBankConflicts32x8(float *odata, const float *idata, int xdim, int ydim) {
-    __shared__ float tile[32][32 + 1];
-    int tilex = blockIdx.x * 32;
-    int tiley = blockIdx.y * 32;
-    int x = tilex + threadIdx.x;
-    int y = tiley + threadIdx.y;
-
-    for (int j = 0; j < 32; j += 8) {
-        int index = (y + j) * xdim + x;
-        if (index < (xdim*ydim)) {
-            tile[threadIdx.y + j][threadIdx.x] = idata[index];
+#include "cuda_utils.h"
+
+namespace iirConvolve2D_Cardinal_BSpline_3_MirrorOffBoundKernels {
+
+const int WARP_SIZE = 32;
+
+const int BLOCK_SIZE = 64; // { 16, 32, 64 }
```

Comment:
I think we have some header file with these constant. If not, we probably should have it :D